### DISCUSSION TITLE
Create an image from the web folder and send it to Docker Desktop

### DISCUSSION THREAD

**Discussion by asemandeveloper** [langgenius/dify#4939] [Suggestion]
I want to just create an image from the web folder where I have made changes to the code and upload it to the Docker Desktop installed on my computer with the "docker compose up -d" command and then create the container and run the project do.
How is this done? please guide me.

In the DIFY project, I have made changes in the codes of the web folder and I want to create an image of the entire project for Docker desktop on windows 11 and run it in docker. What changes should I make in the docker-compose.yaml file of the web service section so that an image can be created from the changed codes in the web folder?

**Comment by patryk20120** [comment]
You can easily build the web image by going into `web` folder and executing `docker build`. Then you can change the image name in docker-compose file to your freshly build image.

**Reply by asemandeveloper** [reply]
I proceeded according to your opinion. I get an error message!
What changes should I make in the docker-compose.yaml file in the docker folder?
What changes should I make in the Dockerfile file of the web folder?
How do I execute docker compose commands?
Please explain step by step.
Thank you

**Reply by patryk20120** [reply]
Well, to preceed you need basic docker knowledge. You can add a tag to your build docker image with `-t` parameter and later on change the docker-compose images to your tag. e.g. `cd web && docker build -t dify-web-custom-image .` will generate an image called `dify-web-custom-image`. Go to docker-compose file and change https://github.com/langgenius/dify/blob/main/docker/docker-compose.yaml#L323 to `dify-web-custom-image:latest`. Than restart the containers.

**Reply by asemandeveloper** [reply]
According to the "Start Local Source Code" guide, I ran the project locally. In this method, API, Worker, WEB services are run independently and outside of Docker Desktop. And other services are first downloaded from Docker Hub and then run on Docker Desktop.
I have made some changes in API, Worker, WEB services code and I want to take them to Docker Desktop and run them. What should I do? please guide me.

By the way, your suggestion that I should do `cd web && docker build -t dify-web-custom-image` does not work.

Please provide a practical solution if possible. As an example, the changes that need to be made in the `docker-compose.yaml`  file to push the API, Worker, WEB (which are now local) to Docker Desktop and then run it alongside other services.

**Reply by patryk20120** [reply]
What do you mean that it does not work? The thing you need to do is to build a new image. That's it.
